loggers/loggers_test.go

```diff
@@ -0,0 +1,89 @@
+package loggers
+
+import (
+	"encoding/json"
+	"math/rand"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+type FakeIoWriter struct {
+	// A list of strings that represent ints
+	ListOfInts []string
+}
+
+func (f *FakeIoWriter) Write(p []byte) (n int, err error) {
+	f.ListOfInts = append(f.ListOfInts, string(p))
+	return len(p), nil
+}
+
+// TestThreadSafetyOfLogger ensures that multiple threads writing to a single source
+// don't get corrupted
+func TestThreadSafetyOfLogger(t *testing.T) {
+	var atomicInt int64 = 0
+
+	fakeWriter := FakeIoWriter{}
+	lMgr := MakeLoggerManager(&fakeWriter)
+
+	const numberOfWritesPerLogger = 20
+	const numberOfLoggers = 15
+
+	var wg sync.WaitGroup
+	wg.Add(numberOfLoggers)
+
+	// Launch go routines
+	for i := 0; i < numberOfLoggers; i++ {
+		go func() {
+			// Sleep a random number of milliseconds before and after to test
+			// that creating a logger doesn't affect thread-safety
+			time.Sleep(time.Duration(rand.Intn(50)) * time.Millisecond)
+			l := lMgr.MakeLogger()
+			l.SetFormatter(&log.JSONFormatter{
+				// We want to disable timestamps to stop logrus from sorting our output
+				DisableTimestamp: true,
+			})
+			time.Sleep(time.Duration(rand.Intn(50)) * time.Millisecond)
```

Comment:
One trick for starting all of the loggers at the same time is to have them all wait on the same wait group. That you mark as Done once the goroutines have been created

Reply:
I want to make sure that they don't start at the same time though.  I want to make sure that making the logger doesn't affect thread-safety.